src/lib.rs

```diff
@@ -29,27 +29,62 @@ pub use crate::node::Taffy;
 #[cfg(feature = "std")]
 use core::fmt::{Display, Formatter, Result};
 
-/// An error that can occur when performing layout
+/// The [`Node`](node::Node) was not found in the [`Taffy`] instance
 #[derive(Debug)]
-pub enum Error {
-    /// The [`Node`](node::Node) was invalid
-    InvalidNode(node::Node),
+pub struct NodeNotFoundError(pub node::Node);
+
+#[cfg(feature = "std")]
+impl Display for NodeNotFoundError {
+    fn fmt(&self, f: &mut Formatter) -> Result {
+        write!(f, "Node {:?} is not in the Taffy instance", self.0)
+    }
+}
+
+#[cfg(feature = "std")]
+impl std::error::Error for NodeNotFoundError {}
+
+/// An error that occurs while trying to access or modify a [`Node`](node::Node)'s children by index.
+#[derive(Debug)]
+pub enum ChildOperationError {
```

Comment:
Hmm. I don't love this name either. Maybe just `ChildError`?

Reply:
Actually: what if we went with `InvalidNode` and `InvalidChild`? And then stuck these in an `error` module, so then users can call them with the path as `error::InvalidNode` if they want?

This gets my vote :)

Reply:
The `error` module idea sounds good. Thank you for the names.